### PR TITLE
test: BI-0 Fix connection base tests to obtain real connection entry from US

### PR DESCRIPTION
**Pull request by khamitovdr**

### CODE REVIEW

File: lib/dl_core_testing/dl_core_testing/connection.py [hunk 1]
```diff
@@ -80,7 +76,13 @@ def make_saved_connection(
 ) -> ConnectionBase:
     conn = make_connection(us_manager=sync_usm, conn_type=conn_type, conn_name=conn_name, data_dict=data_dict)
     sync_usm.save(conn)
-    return conn
+    conn_id = conn.uuid
+    if conn_id is None:
+        raise ValueError("Connection ID is None")
+
+    conn_us = sync_usm.get_by_id(conn_id)
+    assert isinstance(conn_us, ConnectionBase)  # for typing
```
Comment (ovsds) [inline lib/dl_core_testing/dl_core_testing/connection.py:84]: for typing only you can use `typing.cast(...)`

Comment (khamitovdr) [reply lib/dl_core_testing/dl_core_testing/connection.py:84]: Adopting best practices))

https://github.com/datalens-tech/datalens-backend/blob/be2b4de77678102324f62dfbf332c3c37e23cf43/lib/dl_api_lib/dl_api_lib/app/control_api/resources/connections.py#L94

Comment (ovsds) [reply lib/dl_core_testing/dl_core_testing/connection.py:84]: who said it is?

Comment (khamitovdr) [reply lib/dl_core_testing/dl_core_testing/connection.py]: Just a joke)
Fixed!

**Review by ovsds — APPROVED** [review]

**Pull request merged**